### PR TITLE
[GLUTEN-2009][CH] Fix window function lead/lag use null as default value

### PR DESCRIPTION
**Pull request by loneylee**
## What changes were proposed in this pull request?

Fix window function lead/lag use null as default value

Fixes: #2009

## How was this patch tested?

UT coverage.

### CODE REVIEW

**Comment by loneylee** [comment]
Run Gluten Clickhouse CI

File: cpp-ch/local-engine/Parser/WindowRelParser.cpp [hunk 1]
```diff
@@ -315,12 +315,25 @@ void WindowRelParser::tryAddProjectionBeforeWindow(QueryPlan & plan, const subst
...
             const DB::ActionsDAG::Node * node = nullptr;
+
+            if (arg2.has_literal() && arg2.literal().has_null() && !col.type->isNullable())
+            {
+                node = ActionsDAGUtil::convertNodeType(
+                    actions_dag, &actions_dag->findInOutputs(col.name), makeNullable(col.type)->getName(), col.name);
+                actions_dag->addOrReplaceInOutputs(*node);
+                names.emplace_back(node->result_name);
+                types.emplace_back(node->result_type);
+            }
```
Comment (lgbo-ustc) [inline cpp-ch/local-engine/Parser/WindowRelParser.cpp:330]: Is the 3rd arg the default value?
Please add a explainment here

**Comment by lgbo-ustc** [comment]
Others LGTM

File: cpp-ch/local-engine/Parser/WindowRelParser.cpp (hunk 1 above)
Comment (loneylee) [reply cpp-ch/local-engine/Parser/WindowRelParser.cpp:332]: fix

**Review by baibaichen — APPROVED** [review]
LGTM

**Pull request merged**